### ISSUE TITLE
Declare "Static Database Types" for Consumer from XML-File?

### ISSUE THREAD

**Issue by GRieber** [Lawo/ember-plus-sharp#37]
Hello,
is there a chance to automatically build a "Static Database Types" - Declaration with a kind of "Class-Builder" in the Ember-SDK? It would be nice in case of large trees to write manually instead :-)

**Comment by andreashuber-lawo** [comment]
At the moment no such a tool exists. While it would be a worthwhile addition, the implementation might be more difficult than what a casual observer might expect:

1. The user almost never needs to have access to all parameters in a tree. Generating classes for the full tree would therefore often lead to performance issues because the provider is queried for more data than necessary and the additional elements occupy more memory than necessary. The tool would therefore have to offer a way to only generate classes/properties for elements the user is interested in.
2. Different areas of Ember+ trees often contain identical data structures. A naïve implementation would generate multiple classes containing equally-named members. This is problematic if your application needs to apply general algorithms to such data structures. While it is possible to automate this process somewhat, it is certainly not trivial.
3. Ember+ does not distinguish between nodes that act as collections of identically structured elements and nodes that contain elements of differing structure. In a consumer implemented with Ember+ Sharp you'd use [CollectionNode](http://lawo.github.io/ember-plus-sharp/html/BD7E2B1D.htm) for the former and [FieldNode](http://lawo.github.io/ember-plus-sharp/html/5504429E.htm) for the latter. Again, a tool could detect this to a certain degree, but it's almost impossible to automate this perfectly.

TL;DR: Such a tool would certainly help, but it would have to offer many customization points, which is why no attempt has so far been made to implement it.
